source/SH3/system/glprogram.cpp

```diff
@@ -55,6 +55,50 @@ GLint sh3_glprogram::Load(const std::string& name)
     glDeleteShader(fragShader);
 }
 
+GLint sh3_glprogram::Load(const std::string& name, std::vector<std::string> const &attribs)
+{
+    GLint status = 0;
+
+    programID = glCreateProgram();
+
+    // Generate our vertex and fragment shader
+    vertShader = (GLuint)Compile(GL_VERTEX_SHADER);
+    fragShader = (GLuint)Compile(GL_FRAGMENT_SHADER);
+
+    glAttachShader(programID, vertShader);
+    glAttachShader(programID, fragShader);
+
+    // Bind any attributes before we link the program
+    for(int i = 0; i < attribs.size(); i++)
+    {
+        glBindAttribLocation(programID, i, reinterpret_cast<const GLchar*>(attribs[i].c_str()));
```

Comment:
What I meant is that you might accidentally specify an attribute name that does not exist in the shader, for example because of misspelling, or renaming an attribute in the shader source and forgetting recompile the game binary.